ssl/ssl_test.cc

```diff
@@ -6515,9 +6541,9 @@ TEST_P(SSLVersionTest, SSLPending) {
     // fixing either of these bugs, this test may need to be redone.
     EXPECT_EQ(1, SSL_has_pending(client_.get()));
   } else {
-    // In TLS, we do not overread, so |SSL_has_pending| should report no data is
-    // buffered.
-    EXPECT_EQ(0, SSL_has_pending(client_.get()));
+    // In TLS if read ahead is enabled the two records would also have been read
+    // in a single call.
+    EXPECT_EQ(enable_read_ahead(), SSL_has_pending(client_.get()));
```

Comment:
doesn't this depend on the size of the buffer? no read ahead if 2 records can't fit in the buffer, right?

Reply:
If there is no read ahead AWS-LC will read for TLS record header length, parse the message length, then read that exact length only, so the second message will not be read at all.